Crash when returning from screensaver

### Describe the bug

Seems a bit random but mostly happens when the device has been idle for a few minutes. Not specific to the beta afaik.

Only thing necessary to reproduce is to start the app, wait for the android screensaver to start and try to return to the app.


### Logs

---
client: Jellyfin for Android TV
client_version: 0.15.0-beta.4
client_repository: https://github.com/jellyfin/jellyfin-androidtv
type: crash_report
format: markdown
---
### Logs

***Stack Trace***: 
```log
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.jellyfin.androidtv/org.jellyfin.androidtv.ui.browsing.MainActivity}: java.lang.NullPointerException
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3432)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3596)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2067)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7668)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
Caused by: java.lang.NullPointerException
	at org.jellyfin.androidtv.data.querying.StdItemQuery.<init>(StdItemQuery.kt:23)
	at org.jellyfin.androidtv.ui.browsing.BrowseGridFragment.setupQueries(BrowseGridFragment.java:569)
	at org.jellyfin.androidtv.ui.browsing.BrowseGridFragment.onCreate(BrowseGridFragment.java:178)
	at androidx.fragment.app.Fragment.performCreate(Fragment.java:3090)
	at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:475)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:257)
	at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:121)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1433)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2977)
	at androidx.fragment.app.FragmentManager.dispatchCreate(FragmentManager.java:2884)
	at androidx.fragment.app.FragmentController.dispatchCreate(FragmentController.java:252)
	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:220)
	at org.jellyfin.androidtv.ui.browsing.MainActivity.onCreate(MainActivity.kt:38)
	at android.app.Activity.performCreate(Activity.java:8000)
	at android.app.Activity.performCreate(Activity.java:7984)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3405)
	... 11 more

```  
***Logcat***: 
```log
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2977)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at androidx.fragment.app.FragmentManager.dispatchCreate(FragmentManager.java:2884)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at androidx.fragment.app.FragmentController.dispatchCreate(FragmentController.java:252)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:220)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at org.jellyfin.androidtv.ui.browsing.MainActivity.onCreate(MainActivity.kt:38)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.app.Activity.performCreate(Activity.java:8000)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.app.Activity.performCreate(Activity.java:7984)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3405)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3596)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2067)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.os.Handler.dispatchMessage(Handler.java:106)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.os.Looper.loop(Looper.java:223)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at android.app.ActivityThread.main(ActivityThread.java:7668)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at java.lang.reflect.Method.invoke(Native Method)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
12-13 22:37:03.637 E/DisplayPreferencesStore(14888): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
12-13 22:37:03.638 I/DisplayPreferencesStore(14888): Creating an empty DisplayPreferencesDto for next commit.
12-13 22:37:03.639 D/BrowseGridFragment(14888): numCardsScreen <0>
12-13 22:37:03.641 D/AndroidRuntime(14888): Shutting down VM
--------- beginning of crash
12-13 22:37:03.641 E/AndroidRuntime(14888): FATAL EXCEPTION: main
12-13 22:37:03.641 E/AndroidRuntime(14888): Process: org.jellyfin.androidtv, PID: 14888
12-13 22:37:03.641 E/AndroidRuntime(14888): java.lang.RuntimeException: Unable to start activity ComponentInfo{org.jellyfin.androidtv/org.jellyfin.androidtv.ui.browsing.MainActivity}: java.lang.NullPointerException
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3432)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3596)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2067)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.os.Handler.dispatchMessage(Handler.java:106)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.os.Looper.loop(Looper.java:223)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.ActivityThread.main(ActivityThread.java:7668)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at java.lang.reflect.Method.invoke(Native Method)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
12-13 22:37:03.641 E/AndroidRuntime(14888): Caused by: java.lang.NullPointerException
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at org.jellyfin.androidtv.data.querying.StdItemQuery.<init>(StdItemQuery.kt:23)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at org.jellyfin.androidtv.ui.browsing.BrowseGridFragment.setupQueries(BrowseGridFragment.java:569)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at org.jellyfin.androidtv.ui.browsing.BrowseGridFragment.onCreate(BrowseGridFragment.java:178)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at androidx.fragment.app.Fragment.performCreate(Fragment.java:3090)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:475)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:257)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:121)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1433)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2977)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at androidx.fragment.app.FragmentManager.dispatchCreate(FragmentManager.java:2884)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at androidx.fragment.app.FragmentController.dispatchCreate(FragmentController.java:252)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:220)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at org.jellyfin.androidtv.ui.browsing.MainActivity.onCreate(MainActivity.kt:38)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.Activity.performCreate(Activity.java:8000)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.Activity.performCreate(Activity.java:7984)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3405)
12-13 22:37:03.641 E/AndroidRuntime(14888): 	... 11 more
12-13 22:37:03.642 E/ACRA    (14888): ACRA caught a RuntimeException for org.jellyfin.androidtv
12-13 22:37:03.642 E/ACRA    (14888): java.lang.RuntimeException: Unable to start activity ComponentInfo{org.jellyfin.androidtv/org.jellyfin.androidtv.ui.browsing.MainActivity}: java.lang.NullPointerException
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3432)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3596)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2067)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.os.Handler.dispatchMessage(Handler.java:106)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.os.Looper.loop(Looper.java:223)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.ActivityThread.main(ActivityThread.java:7668)
12-13 22:37:03.642 E/ACRA    (14888): 	at java.lang.reflect.Method.invoke(Native Method)
12-13 22:37:03.642 E/ACRA    (14888): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
12-13 22:37:03.642 E/ACRA    (14888): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
12-13 22:37:03.642 E/ACRA    (14888): Caused by: java.lang.NullPointerException
12-13 22:37:03.642 E/ACRA    (14888): 	at org.jellyfin.androidtv.data.querying.StdItemQuery.<init>(StdItemQuery.kt:23)
12-13 22:37:03.642 E/ACRA    (14888): 	at org.jellyfin.androidtv.ui.browsing.BrowseGridFragment.setupQueries(BrowseGridFragment.java:569)
12-13 22:37:03.642 E/ACRA    (14888): 	at org.jellyfin.androidtv.ui.browsing.BrowseGridFragment.onCreate(BrowseGridFragment.java:178)
12-13 22:37:03.642 E/ACRA    (14888): 	at androidx.fragment.app.Fragment.performCreate(Fragment.java:3090)
12-13 22:37:03.642 E/ACRA    (14888): 	at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:475)
12-13 22:37:03.642 E/ACRA    (14888): 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:257)
12-13 22:37:03.642 E/ACRA    (14888): 	at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:121)
12-13 22:37:03.642 E/ACRA    (14888): 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1433)
12-13 22:37:03.642 E/ACRA    (14888): 	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2977)
12-13 22:37:03.642 E/ACRA    (14888): 	at androidx.fragment.app.FragmentManager.dispatchCreate(FragmentManager.java:2884)
12-13 22:37:03.642 E/ACRA    (14888): 	at androidx.fragment.app.FragmentController.dispatchCreate(FragmentController.java:252)
12-13 22:37:03.642 E/ACRA    (14888): 	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:220)
12-13 22:37:03.642 E/ACRA    (14888): 	at org.jellyfin.androidtv.ui.browsing.MainActivity.onCreate(MainActivity.kt:38)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.Activity.performCreate(Activity.java:8000)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.Activity.performCreate(Activity.java:7984)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
12-13 22:37:03.642 E/ACRA    (14888): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3405)
12-13 22:37:03.642 E/ACRA    (14888): 	... 11 more
12-13 22:37:03.642 D/ACRA    (14888): Building report
12-13 22:37:03.642 D/ACRA    (14888): Starting collectors with priority FIRST
12-13 22:37:03.643 D/ACRA    (14888): Calling collector org.acra.collector.DropBoxCollector
12-13 22:37:03.643 D/ACRA    (14888): Calling collector org.acra.collector.LogCatCollector
12-13 22:37:03.643 D/ACRA    (14888): Collector org.acra.collector.DropBoxCollector completed
12-13 22:37:03.643 D/ACRA    (14888): Calling collector org.acra.collector.StacktraceCollector
12-13 22:37:03.643 D/ACRA    (14888): Collector org.acra.collector.StacktraceCollector completed
12-13 22:37:03.645 D/ACRA    (14888): Retrieving logcat output (buffer:default)...

```  

### App information

***App version***: `0.15.0-beta.4` (`150004`)  
***Package name***: `org.jellyfin.androidtv`  
***Build***: 
```json
{"BOARD":"sif","BOOTLOADER":"32.00.2019.50-t210-69ebfcbe","BRAND":"NVIDIA","CPU_ABI":"armeabi-v7a","CPU_ABI2":"armeabi","DEVICE":"sif","DISPLAY":"RQ1A.210105.003.7825199_3167.5736","FINGERPRINT":"NVIDIA\/sif\/sif:11\/RQ1A.210105.003\/7825199_3167.5736:user\/release-keys","HARDWARE":"sif","HOST":"mobile-u64-5306-d6000","ID":"RQ1A.210105.003","IS_DEBUGGABLE":false,"IS_EMULATOR":false,"MANUFACTURER":"NVIDIA","MODEL":"SHIELD Android TV","PERMISSIONS_REVIEW_REQUIRED":true,"PRODUCT":"sif","RADIO":"unknown","SUPPORTED_32_BIT_ABIS":["armeabi-v7a","armeabi"],"SUPPORTED_64_BIT_ABIS":[],"SUPPORTED_ABIS":["armeabi-v7a","armeabi"],"TAGS":"release-keys","TIME":1665975946000,"TYPE":"user","UNKNOWN":"unknown","USER":"NVIDIA","VERSION":{"ACTIVE_CODENAMES":[],"BASE_OS":"","CODENAME":"REL","INCREMENTAL":"7825199_3167.5736","PREVIEW_SDK_FINGERPRINT":"REL","PREVIEW_SDK_INT":0,"RELEASE":"11","RELEASE_OR_CODENAME":"11","SDK":"30","SDK_INT":30,"SECURITY_PATCH":"2022-04-05"}}
```  
***Build config***: 
```json
{"APPLICATION_ID":"org.jellyfin.androidtv","BUILD_TYPE":"release","DEBUG":false,"DEVELOPMENT":false,"VERSION_CODE":150004,"VERSION_NAME":"0.15.0-beta.4"}
```  

### Device information

***Android version***: `11`  
***Device brand***: `NVIDIA`  
***Device product***: `sif`  
***Device model***: `SHIELD Android TV`  

### Crash information

***Start time***: `2022-12-13T22:37:03.353+01:00`  
***Crash time***: `2022-12-13T22:37:03.694+01:00`  



### Application version

0.15.0-beta4

### Where did you install the app from?

Google Play

### Device information

Shield TV

### Android version

11

### Jellyfin server version

10.8.8

--- a/app/src/main/java/org/jellyfin/androidtv/auth/ui/ActivityAuthenticationExtensions.kt
+++ b/app/src/main/java/org/jellyfin/androidtv/auth/ui/ActivityAuthenticationExtensions.kt
@@ -3,6 +3,7 @@ package org.jellyfin.androidtv.auth.ui
 import android.content.Intent
 import androidx.fragment.app.FragmentActivity
 import org.jellyfin.androidtv.auth.repository.SessionRepository
+import org.jellyfin.androidtv.auth.repository.UserRepository
 import org.jellyfin.androidtv.ui.startup.StartupActivity
 import org.koin.android.ext.android.inject
 import timber.log.Timber
@@ -17,8 +18,9 @@ import timber.log.Timber
  */
 fun FragmentActivity.validateAuthentication(): Boolean {
 	val sessionRepository by inject<SessionRepository>()
+	val userRepository by inject<UserRepository>()
 
-	if (sessionRepository.currentSession.value == null) {
+	if (sessionRepository.currentSession.value == null || userRepository.currentUser.value == null) {
 		Timber.w("Activity ${this::class.qualifiedName} started without a session, bouncing to StartupActivity")
 		startActivity(Intent(this, StartupActivity::class.java))
 		finish()

--- a/app/src/main/java/org/jellyfin/androidtv/ui/browsing/MainActivity.kt
+++ b/app/src/main/java/org/jellyfin/androidtv/ui/browsing/MainActivity.kt
@@ -41,7 +41,6 @@ class MainActivity : FragmentActivity(R.layout.fragment_content_view) {
 
 		applyTheme()
 
-
 		backgroundService.attach(this)
 		onBackPressedDispatcher.addCallback(this, backPressedCallback)
 

